tests/unit/recommenders/evaluation/test_spark_evaluation.py

```diff
@@ -501,3 +525,35 @@ def test_serendipity(spark_diversity_data, target_metrics):
         col_relevance="Relevance",
     )
     assert target_metrics["serendipity"] == evaluator.serendipity()
+
+
+@pytest.mark.spark
+def test_user_diversity2(spark_diversity_data, target_metrics):
```

Comment:
I would suggest a more informative name so that it is clear to someone looking at output from pytest that this method uses item features. Similarly for the method below.

Reply:
modified the method name to "test_user_diversity_item_feature_vector", and the same change made for the method below.